src/Microsoft.AspNetCore.Mvc.Core/Internal/ActionResultTypeMapper.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class ActionResultTypeMapper : IActionResultTypeMapper
```

Comment:
Real `internal`

Reply:
I was just about to ask you to make `GetResultDataType` virtual 😝 Why internal?